pkg/generate/app/sourcelookup.go

```diff
@@ -73,19 +73,10 @@ func IsPossibleSourceRepository(s string) bool {
 
 // IsRemoteRepository checks whether the provided string is a remote repository or not
 func IsRemoteRepository(s string) bool {
-	if !s2igit.New(s2iutil.NewFileSystem()).ValidCloneSpecRemoteOnly(s) {
-		return false
-	}
-	url, err := url.Parse(s)
-	if err != nil {
-		return false
-	}
-	url.Fragment = ""
-	gitRepo := git.NewRepository()
-	if _, _, err := gitRepo.ListRemote(url.String()); err != nil {
-		return false
-	}
-	return true
+	// We don't try to connect to the remote registry because we may not yet have
+	// credentials.
+
+	return s2igit.New(s2iutil.NewFileSystem()).ValidCloneSpecRemoteOnly(s)
```

Comment:
Urg, you're right to query this (although it didn't break Cesar's code fwiw).

I was trying to get rid of the ls-remote due to the following scenario: if the git remote requires auth and the user doesn't have .gitcredentials or whatever set up, then because of this function they are prompted for git credentials twice in one new-app invocation.

However, my change as it stands breaks running new-app against a remote URL which returns a template (which we don't seem to have a test case for! I'll add one).  This bit wasn't essential for the PR so I'll revert it.

At the moment, our strategy is effectively your "2." - piggyback on any available local authentication method and print a warning.  I think that's the best approach.

Fixing the "credential prompt twice" issue would clearly be more intrusive than I thought as it becomes impossible to tell the difference between c.SourceRepositories and c.Components.  I'm not sure I want to go there...